### PR TITLE
CreatePolygon

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
New functions added to babylon.mesh.ts, babylon.meshBuilder.ts and babylon.mesh.vertexData.ts.

CreatePolygon and ExtrudePolygon both use PolygonMeshBuilder. The first to create a polygon in the xoz plane and the second to extrude a polygon in y direction. The advantage is that these functions look consistent with existing create mesh functions and sideOrientation is added. Documentation to follow.

### CODE REVIEW

**Comment by BabylonJSGuide** [comment]
Did notice that some shapes such as box, cylinder, sphere etc also had entries in babylon.geometry.ts did not add anything here as I did not know what it did or why it was needed.

**Comment by BabylonJSGuide** [comment]
PS should PR be acceptable and a reference put in whats new please make acknowledgement JohnK not cubees

File: src/Mesh/babylon.mesh.ts [hunk 1]
```diff
@@ -2509,6 +2509,40 @@
         }
 
         /**
+         * Creates a polygon mesh.
+         * Please consider using the same method from the MeshBuilder class instead. 
+         * The polygon's shape will depend on the input parameters and is constructed paralell to a ground mesh.
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.ts]: "parallel"

File: src/Mesh/babylon.mesh.ts [hunk 2]
```diff
@@ -2509,6 +2509,40 @@
...
 
         /**
+         * Creates a polygon mesh.
+         * Please consider using the same method from the MeshBuilder class instead. 
+         * The polygon's shape will depend on the input parameters and is constructed paralell to a ground mesh.
+         * The parameter `shape` is a required array of successive Vector3 representing the corners of the polygon in th XoZ plane, that is y = 0 for all vectors.
+         * You can set the mesh side orientation with the values : BABYLON.Mesh.FRONTSIDE (default), BABYLON.Mesh.BACKSIDE or BABYLON.Mesh.DOUBLESIDE
+         * The mesh can be set to updatable with the boolean parameter `updatable` (default false) if its internal geometry is supposed to change once created.
+         * Remember you can only change the shape positions, not their number when updating a polygon.
+         */
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.ts]: Can you provide a sample PG in the comments here?

File: src/Mesh/babylon.meshBuilder.ts [hunk 3]
```diff
@@ -770,6 +770,54 @@
         }
 
         /**
+         * Creates a polygon mesh.
+         * The polygon's shape will depend on the input parameters and is constructed paralell to a ground mesh.
```
Comment (deltakosh) [inline src/Mesh/babylon.meshBuilder.ts]: "parallel"

File: src/Mesh/babylon.meshBuilder.ts [hunk 4]
```diff
@@ -770,6 +770,54 @@
...
+			var polygonTriangulation = new PolygonMeshBuilder(name, contours, scene);
+			for(var hNb = 0; hNb < holes.length; hNb++) {
+				hole = [];
+				for(var hPoint = 0; hPoint < holes[hNb].length; hPoint++) {
+					hole.push(new Vector2(holes[hNb][hPoint].x, holes[hNb][hPoint].z));
+				}
+				polygonTriangulation.addHole(hole);
+			}
+			var polygon = polygonTriangulation.build(options.updatable, depth);
+            polygon.sideOrientation = options.sideOrientation;
```
Comment (deltakosh) [inline src/Mesh/babylon.meshBuilder.ts]: missing a space at beginning of the line

**Comment by deltakosh** [comment]
Can you also update the what;s new for credits?

**Comment by deltakosh** [comment]
Ok I'll do the merge and fix my comments (please update the what's new still)

**Pull request merged**